### PR TITLE
Add CastLike operator

### PR DESCRIPTION
**Pull request by jantonguirao**
Signed-off-by: Joaquin Anton <janton@nvidia.com>

## Category:

**New feature**

## Description:

- Introduces a new frontend for Cast operator : CastLike
- CastLike applies the type of the second input to the first

### Affected modules and functionalities:

Cast

### Key points relevant for the review:

NA
### Tests:

- [ ] Existing tests apply
- [x] New tests added
  - [x] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [x] Documentation updated
  - [x] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: dali/operators/generic/cast.cc [hunk 1]
```diff
@@ -73,4 +73,11 @@ DALI_SCHEMA(Cast)
     .SupportVolumetric()
     .AddTypeArg("dtype", R"code(Output data type.)code");
 
+DALI_SCHEMA(CastLike)
+    .DocStr("Cast tensor to the type of another tensor.")
```
Comment (stiepan) [inline dali/operators/generic/cast.cc:77]: How about:
```suggestion
    .DocStr("Cast the first tensor to the type of the second tensor.")
```
to make it more explicit which of the inputs gets cast?

**Review by stiepan — APPROVED** [review]

**Review by awolant — APPROVED** [review]
This is a cool new feature, maybe it would make sense to showcase it in one of the tutorials or examples in the docs.

**Comment by jantonguirao** [comment]
!build

**Review by awolant — APPROVED** [review]

**Comment by jantonguirao** [comment]
!build

**Comment by jantonguirao** [comment]
!build

**Comment by jantonguirao** [comment]
!build

**Pull request merged**